src/Compilers/CSharp/Portable/Symbols/Source/SourceFieldLikeEventSymbol.cs

```diff
@@ -239,6 +253,96 @@ internal SourceEventDefinitionAccessorSymbol(
+
+            protected override SourceMemberMethodSymbol? BoundAttributesSource
+            {
+                get
+                {
+                    return IsExtern && this.MethodKind == MethodKind.EventAdd
+                        ? (SourceMemberMethodSymbol?)this.AssociatedEvent.RemoveMethod
+                        : null;
+                }
+            }
+
+            protected override IAttributeTargetSymbol AttributeOwner
+            {
+                get
+                {
+                    Debug.Assert(PartialDefinitionPart is null);
+
+                    switch (PartialImplementationPart)
+                    {
+                        case SourceCustomEventAccessorSymbol:
+                            return this;
+
+                        case SynthesizedEventAccessorSymbol:
+                            Debug.Assert(IsExtern);
+                            return AssociatedEvent;
+
+                        case null:
+                            // Might happen in error scenarios.
+                            return this;
+
+                        default:
+                            Debug.Assert(false);
+                            return this;
+                    }
+                }
+            }
+
+            internal override OneOrMany<SyntaxList<AttributeListSyntax>> GetAttributeDeclarations()
+            {
+                Debug.Assert(PartialDefinitionPart is null);
+
+                switch (PartialImplementationPart)
+                {
+                    case SourceCustomEventAccessorSymbol customImplementationPart:
+                        return OneOrMany.Create(customImplementationPart.AttributeDeclarationSyntaxList);
+
+                    case SynthesizedEventAccessorSymbol synthesizedImplementationPart:
+                        Debug.Assert(IsExtern);
+                        return OneOrMany.Create(
+                            AssociatedEvent.AttributeDeclarationSyntaxList,
+                            synthesizedImplementationPart.AssociatedEvent.AttributeDeclarationSyntaxList);
+
+                    case null:
+                        // Might happen in error scenarios.
+                        return OneOrMany<SyntaxList<AttributeListSyntax>>.Empty;
+
+                    default:
+                        Debug.Assert(false);
+                        return OneOrMany<SyntaxList<AttributeListSyntax>>.Empty;
+                }
+            }
+
+            internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<CSharpAttributeData> attributes)
+            {
+                Debug.Assert(PartialDefinitionPart is null);
+
+                if (PartialImplementationPart is { } implementationPart)
+                {
+                    implementationPart.AddSynthesizedAttributes(moduleBuilder, ref attributes);
+                }
+                else
+                {
+                    base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
```

Comment:
>base.AddSynthesizedAttributes(moduleBuilder, ref attributes); [](http://example.com/codeflow?start=20&length=61)

When do we hit this case?

Reply:
In error scenarios when the partial implementation part is missing, but the `event` is partial.